### DISCUSSION TITLE
Merge overlaid las files

### DISCUSSION THREAD

**Discussion by seanreilly66** [r-lidar/lidR#737] [Q&A] [answered]
Hi everyone,

I have two registered las files that I am attempting to merge. Is there a recommended way to do this within the lidR package? Currently, I have come up with the following but this method is slow for pairs of large tls las files (>3 gb):

```r

library(lidR)

las1 <- LAS(data = data.frame(
  X = runif(100, 1, 10),
  Y = runif(100, 1, 10),
  Z = runif(100, 1, 10)
))

las2 <- LAS(data = data.frame(
  X = runif(100, 1, 10),
  Y = runif(100, 1, 10),
  Z = runif(100, 20, 30)
))

merge_df <- rbind(las1@data, las2@data)

merge_las <- LAS(merge_df)

```

Thank you! I really appreciate any suggestions.

Sean

**Comment by Jean-Romain** [comment, marked as answer]
You can use the `catalog_retile` being clever on how to make chunks

```r
ctg = readLAScatalog(...)
opt_chunk_size(ctg) = 2000 #large enough to encompass your tiles
opt_chunk_alignment(ctg) = c(...) # may be required to make only one chunk
plot(ctg, chunk = T) # to check if there is only one chunk
new_ctg = catalog_retile(ctg)
```

This should use very little memory. Yet the output file will be large.

You can also have a look to my new package `lasR`. There is merge file section in the tutorial https://r-lidar.github.io/lasR/articles/lasR2.html#write-las

**Reply by seanreilly66** [reply]
Thank you for the reply and everything you've done with lidR. This code worked great and I will definitely give lasR a try.

**Reply by seanreilly66** [reply]
Is there a way to feed the output from this (e.g., new_ctg) into cloud_metrics without first re-reading from disk? I apologize if there is an answer to this already elsewhere on this page but I have not been able to find it.

**Reply by Jean-Romain** [reply]
Nope, `cloud_metric()` calls a user defined R function to your point cloud and thus requires your point cloud to be loaded in memory in a `data.frame`. `catalog_retile()` is full c++ and is capable of retiling without loading the point cloud in R memory. Both a very incompatible.
